widget/ui/src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,77 @@
+import type { PropTypes } from './Tabs.types';
+
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Divider } from '../Divider';
+import { Tooltip } from '../Tooltip';
+
+import { BackdropTab, Tab, Tabs } from './Tabs.styles';
+
+export function TabsComponent(props: PropTypes) {
+  const {
+    items,
+    onChange,
+    container = document.body,
+    value,
+    currentTabIndex,
+    type,
+    borderRadius,
+  } = props;
+  const [tabWidth, setTabWidth] = useState(0);
+  const tabRef: React.Ref<HTMLButtonElement> = useRef(null);
+
+  useEffect(() => {
+    const updateTabWidth = () => {
+      if (tabRef.current) {
+        const tabRect = tabRef.current.getBoundingClientRect();
+        setTabWidth(tabRect.width);
+      }
+    };
+
+    updateTabWidth();
+    window.addEventListener('resize', updateTabWidth);
+
+    return () => {
+      window.removeEventListener('resize', updateTabWidth);
+    };
+  }, []);
+  return (
+    <Tabs type={type} borderRadius={borderRadius}>
+      {items.map((item, index) => (
+        <Tooltip
```

Comment:
This Tooltip component has caused a problem which is observable in the below image: 

<img width="640" alt="image" src="https://github.com/rango-exchange/rango-client/assets/154742563/e26d2faf-73b8-4aa8-978a-0fa3bca519bc">

Reply:
it is related to the logic of this line of code:
"open={!item.tooltip || undefined}"

By the way, the "open" prop seems a bit tricky. It would be helpful to add a comment explaining its functionality for better understanding and clarity.

Reply:
fixed